src/rprblender/engine/viewport_engine.py

```diff
@@ -349,7 +356,17 @@ class FinishRender(Exception):
 
                     # checking for last iteration
                     # preparing information to show in viewport
+                    time_render_prev = time_render
                     time_render = time.perf_counter() - time_begin
+                    iteration_time = time_render - time_render_prev
+                    if depsgraph.scene.rpr.viewport_limits.adapt_viewport_resolution:
+                        if iteration == 2:
+                            target_time = 1.0 / depsgraph.scene.rpr.viewport_limits.viewport_samples_per_sec
+                            render_pixels = self.rpr_context.width * self.rpr_context.height
+                            self.adapt_render_pixels = render_pixels * target_time / iteration_time
```

Comment:
I think it would be more clear if passing the ratio target_time / iteration_time to adapt_resolution instead of the adapt_render_pixels.